### PR TITLE
Add submodule to coreMQTT repository

### PR DESCRIPTION
**Pull request by aggarw13**
Add submodule reference to `v1.0.0` tag of [coreMQTT](https://github.com/FreeRTOS/coreMQTT) repository

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by leegeth — COMMENTED** [review]
This coreMQTT submodule currently is pointing to latest commit on master. I believe we will have to use the released tag instead of master.

**Review by leegeth — DISMISSED** [review]

**Review by gkwicker — APPROVED** [review]

**Comment by aggarw13** [comment]
>This coreMQTT submodule currently is pointing to latest commit on master. I believe we will have to use the released tag instead of master.

Thanks for pointing it. I have updated the commit to point the [v1.0.0](https://github.com/FreeRTOS/coreMQTT/commits/v1.0.0) released tag

File: .gitmodules [hunk 1]
```diff
@@ -30,3 +30,6 @@
 [submodule "libraries/abstractions/pkcs11/psa"]
 	path = libraries/abstractions/pkcs11/psa
 	url = https://github.com/Linaro/freertos-pkcs11-psa.git
+[submodule "libraries/c_sdk_v4/standard/coreMQTT"]
+	path = libraries/c_sdk_v4/standard/coreMQTT
+	url = git@github.com:FreeRTOS/coreMQTT.git
```
Comment (sarenameas) [inline .gitmodules:35]: This needs to be a https path

**Review by leegeth — APPROVED** [review]

**Review by sarenameas — APPROVED** [review]

**Review by gkwicker — APPROVED** [review]

**Pull request merged**